src/WebJobs.Script/Description/Workers/FunctionInvocationCanceledException.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Host;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public class FunctionInvocationCanceledException : FunctionInvocationException
```

Comment:
Have we considered using or deriving from `OperationCanceledException`? I recommend we consider it because it is an established pattern to use that exception type (or something derived from it) to represent cancellation. In fact, many libraries and tools - even the BCL - has special case handling in places for this exception type. For example, a `Task` representing work which throws this exception becomes "canceled" and not "failed".

Reply:
I think we want to use `FunctionInvocationException` because a) we want to link this to a function invocation and make that clear to users and b) how the exception middleware handles exceptions that are / are not `FunctionInvocationException` (though we can always change this if there's a strong enough argument)

Reply:
I'm thinking about where we want to have this defined, in case we want this to flow all the way back up to WebJobs.